### PR TITLE
Deprecated: Use ptr. To instead

### PR DESCRIPTION
**Pull request by yt-huang**
What type of PR is this?

/kind cleanup

What this PR does / why we need it:

https://pkg.go.dev/k8s.io/utils/pointer#BoolPtr

BoolPtr is a function variable referring to Bool.
Deprecated: Use ptr. To instead.
for back-compat
This composite literal allocates a new struct instance with the given values.
You may either specify the field values in order or use their names.

Which issue(s) this PR fixes:
None

Special notes for your reviewer:
Does this PR introduce a user-facing change?
NONE
Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
NONE

### CODE REVIEW

**Comment by cici37** [comment]
/remove-sig api-machinery
Hi @yt-huang, Thanks for the pr! Would you mind adding the related issue into the pr description if there is any? Thanks!

File: pkg/api/service/testing/make.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/intstr"
 	utilpointer "k8s.io/utils/pointer"
+	"k8s.io/utils/ptr"
```
Comment (danwinship) [inline pkg/api/service/testing/make.go:23]: you should get rid of all of the uses of `utilpointer` in the files you're touching

Comment (yt-huang) [reply pkg/api/service/testing/make.go]: I have replaced all instances of 'pointer' with 'ptr and The code has been re-committed.

**Comment by danwinship** [comment]
/lgtm
/approve
(for the directories I can approve)

note that it's sometimes easier to get PRs like this merged if you split them up into separate PRs per SIG...

**Review by Octopusjust — COMMENTED** [review]
/lgtm
/assign @deads2k

**Pull request closed without merging**